torch_geometric/nn/conv/ker_gnn_conv.py

```diff
@@ -0,0 +1,138 @@
+
+    Args:
+        in_channels (int): Size of each input sample, or :obj:`-1` to derive
+            the size from the first input(s) to the forward method.
+        out_channels (int): Size of each output sample.
+        hidden_channels (int): Hidden node feature dimensionality. (default: :obj:`None`)
+        kernel (str): Kernel type. (default: :obj:`rw`)
+        power (int): The power of the adjacency matrix to use. (default: :obj:`1`)
+        size_graph_filter (int):
+        size_subgraph (int):
+        dropout (float, optional): Dropout probability of the nodes during training. (default: 0)
+        **kwargs (optional): Additional arguments of
+            :class:`torch_geometric.nn.conv.MessagePassing`.
+
+    Shapes:
+        - **input:**
+          node features :math:`(|\mathcal{V}|, F_{in})`,
+          edge indices :math:`(2, |\mathcal{E}|)`
+        - **output:**
+          node features :math:`(|\mathcal{V}|, |P| \cdot F_{out})`
+    """
+    def __init__(self, in_channels: int, out_channels: int,
+                 hidden_channels: int = None, kernel: str = 'rw',
+                 power: int = 1, size_graph_filter: int = 6,
+                 size_subgraph: int = 10, dropout: float = 0., **kwargs):
+        kwargs.setdefault('aggr', 'add')
+        super().__init__(**kwargs)
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.hidden_channels = hidden_channels
+        self.kernel = kernel
+        self.power = power
+        self.size_graph_filter = size_graph_filter
+        self.size_subgraph = size_subgraph
+
+        if hidden_channels:
+            self.linear = Linear(in_channels, hidden_channels)
+            self.filter_x = torch.empty(size_graph_filter, hidden_channels,
+                                        out_channels)
+        else:
+            self.filter_x = torch.empty(size_graph_filter, in_channels,
+                                        out_channels)
+
+        self.filter_edge_index = torch.triu_indices(size_graph_filter,
+                                                    size_graph_filter, 1)
+        self.filter_edge_attr = torch.empty(
+            size_graph_filter * (size_graph_filter - 1) // 2, out_channels)
+        self.filter_g = None
+
+        self.simple_conv = SimpleConv()
+        self.dropout = nn.Dropout(p=dropout)
+        self.relu = nn.ReLU()
+        self.ker_linear = Linear(
+            out_channels,
+            out_channels) if kernel == 'drw' else torch.nn.Identity()
+
+        self.reset_parameters()
+        assert self.filter_g is not None and self.filter_g.validate()
+
+    def reset_parameters(self):
+        super().reset_parameters()
+        self.filter_x.data.uniform_(0, 1)
+        self.filter_edge_attr.data.uniform_(-1, 1)
+
+        self.filter_edge_index, self.filter_edge_attr = to_undirected(
+            self.filter_edge_index, self.filter_edge_attr)
+
+        self.filter_g = Data(x=self.filter_x,
```

Comment:
Can we register `filter_x`, `filter_edge_index`, and `filter_edge_attr` as constants for the module using `register_buffer`? This should avoid the `to(device)` operation.